File: FactoryDesignPattern/Shapes_Example_Python/Shapes/Rectangle.py
from Shapes.Shape import Shape

class Rectangle(Shape):

  def __init__(self, height: float, width: float):
    self.height = height
    self.width = width

  def area(self) -> float:
    return self.height * self.width

  def describe(self) -> None:
    print(f"This is a Rectangle with Height: {self.height}, and Width:{self.width} with Area:{self.area()}")

File: FactoryDesignPattern/Shapes_Example/Rectangle.java
package FactoryDesignPattern.Shapes_Example;

public class Rectangle extends Shape {
  private double width;
  private double height;

  public Rectangle(double width, double height) {
    this.width = width;
    this.height = height;
  }

  @Override
  protected double area() {
    return this.width * this.height;
  }

  @Override
  public void describe() {
    System.out.printf("This is a Rectangle with area: %.2f%n", area());
  }
}
